File: framework/languages/java/samples/java/TicTacToe.SessionGateway/src/main/java/systems/zlink/samples/tictactoe/sessiongateway/server/api/handlers/AuthenticateActorHandler.java
package systems.zlink.samples.tictactoe.sessiongateway.server.api.handlers;

import java.util.concurrent.CompletableFuture;
import java.util.concurrent.CompletionStage;
import systems.zlink.framework.channels.ZLinkRequestContext;
import systems.zlink.framework.channels.ZLinkRequestHandler;

public final class AuthenticateActorHandler implements ZLinkRequestHandler<String, String> {
    @Override
    public CompletionStage<String> handleAsync(String request, ZLinkRequestContext context) {
        if (!request.startsWith("player-")) {
            throw new IllegalArgumentException("unknown actor token");
        }
        return CompletableFuture.completedFuture(request);
    }
}

File: framework/languages/java/samples/kotlin/TicTacToe.SessionGateway/src/main/kotlin/systems/zlink/samples/kotlin/tictactoe/sessiongateway/server/api/handlers/AuthenticateActorHandler.kt
package systems.zlink.samples.kotlin.tictactoe.sessiongateway.server.api.handlers

import java.util.concurrent.CompletableFuture
import java.util.concurrent.CompletionStage
import systems.zlink.framework.handlers.ZLinkHandlerGroup
import systems.zlink.framework.handlers.ZLinkRequest

@ZLinkHandlerGroup("api")
class AuthenticateActorHandler {
    @ZLinkRequest(packetName = "AuthenticateActor")
    fun handleAsync(request: String): CompletionStage<String> {
        require(request.startsWith("player-")) { "unknown actor token" }
        return CompletableFuture.completedFuture(request)
    }
}
